UKHomeOffice/sgmr-service
# NMSW-1370 Upgrade to React 18 & latest testing library packages

## AC

To enable use of more modern features & other packages

requirements

- update react 
- update react-dom

potential peer dependency issues

- react testing library - updated in this PR
- reach autocomplete - has been removed

----

## To test

- remove node_modules
- `npm ci`
- `npm test`
- `npm run lint`
- `npm start`
- > all should pass and app should run as normal

----

## Notes

Needed to update some tests with the upgrade to testing-library. As part of this I updated some of the waits to the findBy way, and some of the getBy's into getByRole to bring them in line with how we like to do testing now
